SpringProject/src/it/java/com/gradproject2019/userConference/UserConferenceControllerIT.java

```diff
@@ -0,0 +1,85 @@
+package com.gradproject2019.userConference;
+
+import com.gradproject2019.userConference.data.UserConferenceRequestDto;
+import com.gradproject2019.userConference.data.UserConferenceResponseDto;
+import com.gradproject2019.userConference.persistance.UserConference;
+import com.gradproject2019.utils.TestUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static com.gradproject2019.userConference.data.UserConferenceRequestDto.UserConferenceRequestDtoBuilder.anUserConferenceRequestDto;
+import static org.springframework.http.HttpMethod.POST;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class UserConferenceControllerIT extends TestUtils {
+    @LocalServerPort
+    int testServerPort;
+
+    private String baseUri;
```

Comment:
Could just change this to a URI instead of a string

Reply:
If we use a URI here we cannot concatenate the baseUri to include the testServerPort, is there another method for doing this?